interface/src/app/node/node.js

```diff
@@ -304,6 +307,32 @@ angular.module('adage.node', [
         $scope.numExpShown =
           $scope.topMode ? $scope.topNum : $scope.experiments.length;
       };
+
+      // Event handler when user clicks the checkbox in front of each sample.
+      // It keeps track of the number of selected samples in each experiment
+      // to indicate whether the "show annotations" button should be enabled.
+      $scope.toggleSelection = function(exp, selected) {
+        if (selected) {
+          ++exp.numSelections;
+        } else {
+          --exp.numSelections;
+        }
+      };
```

Comment:
@dongbohu I figured that performance was partly why you did it this way. I was completely surprised to find this kind of looping to be *very* fast when I was checking on the performance of the javascript in the heatmap. I fully expected a crazy loop I had written to be the source of some slow performance I observed and the code completely zipped right through a more complex loop over 600 nodes * 10 samples (so about 6000 entries) in a complex data structure.